src/app/base/components/DhcpFormFields/DhcpFormFields.tsx

```diff
@@ -134,21 +131,21 @@ export const DhcpFormFields = ({ editing }: Props): JSX.Element => {
           { value: "device", label: "Device" },
         ]}
       />
-      {type &&
+      {type === "machine" ? (
+        <FormikField component={MachineSelect} name="entity" />
```

Comment:
Is this what you had in mind @huwshimi ?

Reply:
Yep! And then you need to wrap the contents in a `<Field>` component (here's a good example: https://github.com/canonical/maas-ui/blob/main/src/app/base/components/TagSelector/TagSelector.tsx#L250).

Reply:
Done